### PR TITLE
Add support for emitting static properties in class expressions in ES6.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/Microsoft/TypeScript/issues/2574

### CODE REVIEW

File: tests/baselines/reference/classExpressionWithStaticPropertiesES61.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+//// [classExpressionWithStaticPropertiesES61.ts]
+var v = class C { static a = 1; static b = 2 };
+
+//// [classExpressionWithStaticPropertiesES61.js]
+var v = (_a = class C {
```
Comment (mhegazy) [inline tests/baselines/reference/classExpressionWithStaticPropertiesES61.js]: can you add a test with a decorator as well.

Comment (mhegazy) [inline tests/baselines/reference/classExpressionWithStaticPropertiesES61.js]: I think the current approach should work.
```ts
_a = class C {} ,
_a = __decorate(C, dec),
_a.a = 1,
_a;
```

**Comment by mhegazy** [comment]
:+1:

File: src/compiler/emitter.ts [hunk 2]
```diff
@@ -3590,9 +3638,24 @@ module ts {
                 // From ES6 specification:
                 //      HasLexicalDeclaration (N) : Determines if the argument identifier has a binding in this environment record that was created using
                 //                                  a lexical declaration such as a LexicalDeclaration or a ClassDeclaration.
-                writeLine();
-                emitMemberAssignments(node, NodeFlags.Static);
-                emitDecoratorsOfClass(node);
+
+                if (isClassExpressionWithStaticProperties) {
+                    for (var property of staticProperties) {
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: let

**Pull request merged**